ui/panels-plugin/src/plugins/time-series-chart/PanelComponent.stories.tsx

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { Meta, StoryObj } from '@storybook/react';
+import { TimeSeriesChart } from '@perses-dev/panels-plugin';
+import {
+  WithDataQueries,
+  WithPluginRegistry,
+  WithTimeRange,
+  WithTemplateVariables,
+  WithDatasourceStore,
+} from '@perses-dev/plugin-system/src/stories/shared-utils';
+import { mockTimeSeriesResponseWithStableValue } from '@perses-dev/internal-utils';
+import { mockQueryRangeRequests, waitForStableCanvas, WithQueryClient, WithQueryParams } from '@perses-dev/storybook';
+import { ComponentProps } from 'react';
+
+type TimeSeriesChartProps = ComponentProps<typeof TimeSeriesChart.PanelComponent>;
+
+type TimeSeriesChartWrapperProps = TimeSeriesChartProps & {
+  height?: number;
+  width?: number;
+};
+function TimeSeriesChartWrapper({ height, width, ...otherProps }: TimeSeriesChartWrapperProps) {
+  return (
+    <div style={{ width: width, height: height }}>
+      <TimeSeriesChart.PanelComponent {...otherProps} />
+    </div>
+  );
+}
+
+const TIMESERIES_EXAMPLE_MOCK_NOW = 1673805600000;
+const TIMESERIES_EXAMPLE_MOCK_START = TIMESERIES_EXAMPLE_MOCK_NOW - 6 * 60 * 60 * 1000;
+
+/**
+ * The panel component for the `TimeSeriesChart` plugin.
+ *
+ * This component is not intended to be used directly. It is documented in storybook
+ * to provide an example of what the plugin panel component looks like.
+ */
+const meta: Meta<typeof TimeSeriesChart.PanelComponent> = {
+  component: TimeSeriesChart.PanelComponent,
+  argTypes: {},
+  parameters: {
+    WithDataQueries: {
```

Comment:
Does this need to be lower camel case? (like `withTimeRange`)

Reply:
Apparently this one is upper camel. Looks like we have some inconsistences in those, which isn't ideal. I really wish storybook had better type integration for parameters, so we'd catch this stuff more easily. I'm going to leave this alone in this PR and do a separate PR to fix the naming inconsistencies in the decorators.